arch/x86/core/pcie.c

```diff
@@ -155,19 +157,144 @@ void pcie_conf_write(pcie_bdf_t bdf, unsigned int reg, uint32_t data)
 
 #ifdef CONFIG_PCIE_MSI
 
+#ifdef CONFIG_INTEL_VTD_ICTL
+
+#include <drivers/interrupt_controller/intel_vtd.h>
+
+static const struct device *vtd;
+
+static bool get_vtd(void)
+{
+	if (vtd != NULL) {
+		return true;
+	}
+
+	vtd = device_get_binding("VTD_0");
```

Comment:
Better to use `DT_*` macro here for device name.

Reply:
will do